### ISSUE TITLE
Fix `stylelint-disable-next-line` disables autofix for entire file

### ISSUE THREAD

**Issue by SoundOfTheSky** [stylelint/stylelint#6583]
### What steps are needed to reproduce the bug?

Disable rule for one line somewhere in one file.
Example: 
```
.my-block {
  <- Will not be autofixed
  width: 160px;

  /* stylelint-disable-next-line declaration-empty-line-before */
  height: 160px;
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-empty-line-before": [
      "always",
      {
        "except": ["after-declaration", "first-nested"],
        "ignore": ["after-comment", "inside-single-line-block"]
      }
    ]
  }
}
```

### How did you run Stylelint?

npx stylelint "src/**/*.{css,scss,sass}" --fix   

### Which version of Stylelint are you using?

`14.16.1`

### What did you expect to happen?

Code above autofixed to this:
```
.my-block {
  width: 160px;

  /* stylelint-disable-next-line declaration-empty-line-before */
  height: 160px;
}
```

### What actually happened?

The following problems were reported:

```
test.scss
3:3  ✖  Unexpected empty line before declaration  declaration-empty-line-before
```

### Does the bug relate to non-standard syntax?

No
